File: basic-java-app/app/src/main/java/com/driouechemed/basicjavaapp/repositories/TaskRepository.java
package com.driouechemed.basicjavaapp.repositories;

import com.driouechemed.basicjavaapp.database.AppDatabase;
import com.driouechemed.basicjavaapp.database.entities.Task;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Observable;

public class TaskRepository {

    private final AppDatabase appDatabase;

    public TaskRepository(AppDatabase appDatabase) {
        this.appDatabase = appDatabase;
    }

    public Completable insert(Task task) {
        return appDatabase.getTaskDao().insertTask(task);
    }

    public Observable<List<Task>> getAllTasks() {
        return appDatabase.getTaskDao().getAllTasks();
    }
}

File: basic-kotlin-app/app/src/main/java/com/driouechemed/basickotlinapp/repositories/TaskRepository.kt
package com.driouechemed.basickotlinapp.repositories

import com.driouechemed.basickotlinapp.database.AppDatabase
import com.driouechemed.basickotlinapp.database.entities.Task
import kotlinx.coroutines.flow.Flow

class TaskRepository(private val appDatabase: AppDatabase) {

    suspend fun insert(task: Task) {
        return appDatabase.taskDao().insert(task)
    }

    fun getAll(): Flow<List<Task>> {
        return appDatabase.taskDao().getAll()
    }
}